diff --git a/node_modules/@react-navigation/native/lib/module/index.js b/node_modules/@react-navigation/native/lib/module/index.js
index c596c2b..737a54e 100644
--- a/node_modules/@react-navigation/native/lib/module/index.js
+++ b/node_modules/@react-navigation/native/lib/module/index.js
@@ -11,5 +11,6 @@ export { default as useLinkBuilder } from './useLinkBuilder';
 export { default as useLinkProps } from './useLinkProps';
 export { default as useLinkTo } from './useLinkTo';
 export { default as useScrollToTop } from './useScrollToTop';
+export { default as stripKeysFromNavigationState } from './stripKeysFromNavigationState';
 export * from '@react-navigation/core';
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js b/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js
new file mode 100644
index 0000000..31ba628
--- /dev/null
+++ b/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js
@@ -0,0 +1,17 @@
+// This function removes keys and some other properties from the navigation state 
+// in order to force react-navigation to regenerate those keys
+
+// keys - if we try to use the same keys it is going to break the navigation
+// stale - with stale: false navigation won't regenerate the state 
+// routeNames - to be sure that we have updated routeNames if something changed 
+
+// The argument for this function may be: NavigationState, NavigatoinRoute, PartialState<NavigationState>
+const stripKeysFromNavigationState = ({key, stale, routeNames, routes, state, ...rest}) => {
+  return {
+    ...rest, 
+    routes: routes && routes.map(stripKeysFromNavigationState),
+    state: state && stripKeysFromNavigationState(state)
+  }
+}
+
+export default stripKeysFromNavigationState;
\ No newline at end of file
diff --git a/node_modules/@react-navigation/native/lib/module/useLinking.js b/node_modules/@react-navigation/native/lib/module/useLinking.js
index ee1a4a5..a83aa0d 100644
--- a/node_modules/@react-navigation/native/lib/module/useLinking.js
+++ b/node_modules/@react-navigation/native/lib/module/useLinking.js
@@ -1,4 +1,5 @@
 import { findFocusedRoute, getActionFromState as getActionFromStateDefault, getPathFromState as getPathFromStateDefault, getStateFromPath as getStateFromPathDefault } from '@react-navigation/core';
+import stripKeysFromNavigationState from './stripKeysFromNavigationState';
 import isEqual from 'fast-deep-equal';
 import * as React from 'react';
 import createMemoryHistory from './createMemoryHistory';
@@ -171,7 +172,16 @@ export default function useLinking(ref, _ref) {
       const record = history.get(index);
 
       if ((record === null || record === void 0 ? void 0 : record.path) === path && record !== null && record !== void 0 && record.state) {
-        navigation.resetRoot(record.state);
+        const rootState = navigation.getRootState();
+        // The top level key of current state and cached state may be different.
+        // That will happen if the user switched between wide and narrow view,
+        // because we are using different navigators for these views.
+        // In that case we need to regenerate the state with new keys to make navigation work 
+        if (rootState.key !== record.state.key) {
+          navigation.resetRoot(stripKeysFromNavigationState(record.state));
+        } else {
+          navigation.resetRoot(record.state);
+        } 
         return;
       }
 
